dask_expr/_merge.py

```diff
@@ -352,6 +357,45 @@ def _lower(self):
         return BlockwiseMerge(left, right, **self.kwargs)
 
     def _simplify_up(self, parent, dependents):
+        if isinstance(parent, Filter):
+            new_left = self.left
+            new_right = self.right
+            predicate_cols = set()
+            predicate = parent.predicate
+            if isinstance(predicate, (Projection, Unaryop)):
+                predicate_cols = set(predicate.columns)
+            elif isinstance(predicate, Binop):
+                # FIXME: This doesn't work yet. The optimizer drops the outer
+                # filter
+                # if isinstance(predicate, And):
+                #     new = Filter(self, predicate.left)
+                #     new_pred = predicate.right.substitute(self, new)
+                #     return Filter(new, new_pred)
```

Comment:
Maybe, I'm just doing it wrong but this breaks everything. Maybe @phofl can tell me why this is wrong

Reply:
Yeah I figured out what's wrong, I'll explain offline though, not sure if I can communicate this well this late